### ISSUE TITLE
Dynamic length definition is not supported

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#138]
The code below will fail to be parsed but this should be supported. 

```fortran 
PROGRAM p1
  CALL sub1
END PROGRAM p1

SUBROUTINE sub1(data_length)
  INTEGER, INTENT(IN) :: data_length
  CHARACTER(LEN=data_length+2) :: data

END SUBROUTINE
```

```bash
"prog1.f90", line 7: length specification must be positive
"prog1.f90", line 7: Invalid type spec
```

**Comment by clementval** [comment]
@shingo-s @h-murai Looks like it is not reproducible all the time. If I try to parse this code 10 times, 5 times it will pass without a problem and 5 times it will fail with the error.

**Comment by clementval** [comment]
@shingo-s @h-murai 
Attempt to fix this issue in PR  #145
